android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt

```diff
@@ -97,26 +97,7 @@ class GeckoSitePermissionsStorage(
             onDiskStorage.findSitePermissionsBy(origin, private = private)
         val geckoPermissions = findGeckoContentPermissionBy(origin, includeTemporary, private).groupByType()
 
-        return mergePermissions(onDiskPermission, geckoPermissions)?.resetNoGeckoPermissionIfNeeded(
-            private,
-        )
-    }
-
-    @VisibleForTesting
-    internal fun SitePermissions?.resetNoGeckoPermissionIfNeeded(private: Boolean): SitePermissions? {
-        return if (private) {
-            /**
-             * As part of the GeckoView/AC site permissions migration,
-             * we ended up with a fragmented permissions storage in AC.
-             * Where some permissions are store directly by gecko and others
-             * store locally in AC as GeckoView doesn't offer storage support for them.
-             * The permissions below are stored and manged by AC.
-             * For more references see https://bugzilla.mozilla.org/show_bug.cgi?id=1796614
-             */
-            this?.copy(camera = NO_DECISION, microphone = NO_DECISION, bluetooth = NO_DECISION)
-        } else {
-            this
-        }
+        return mergePermissions(onDiskPermission, geckoPermissions)
     }
```

Comment:
This is not needed anymore, as we are doing [it on OnDiskSitePermissionsStorage](https://github.com/mozilla-mobile/firefox-android/pull/212/files#diff-efeebe6e994fdace8f8983a2afe13d8a43e08abd359921dd27472d0b00747c53R92)